src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingNullCaseInSwitchCheck.java

```diff
@@ -0,0 +1,195 @@
+ * </p>
+ * <p>
+ * Rationale: switch statements and expressions in Java throw a
+ * {@code NullPointerException} if the selector expression evaluates to {@code null}.
+ * With the introduction of patterns in case label,
+ * It is now possible to integrate a null check within the switch,
+ * eliminating the risk of {@code NullPointerException} and simplifies the code
+ * as there is no need for an external null check before entering the switch.
+ * </p>
+ * <p>
+ * Limitations due to not being type-aware:
+ * </p>
+ * <ul>
+ * <li>
+ * This check does not differentiate between primitive types and
+ * their boxed counterparts in case labels.
+ * Therefore, any case label with an integer will not violate this check,
+ * even if it is a reference type {@code Integer}.
+ * </li>
+ * <li>
+ * This check will not violate case labels with idents even if
+ * this is an identifier of Enums. We can't tell if the identifier
+ * is an Enum or just a normal final variable.
+ * </li>
+ * </ul>
+ * <p>
+ * See the <a href="https://docs.oracle.com/javase/specs/jls/se22/html/jls-15.html#jls-15.28">
+ * Java Language Specification</a> for more information about switch statements and expressions.
+ * </p>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code missing.switch.nullcase}
+ * </li>
+ * </ul>
+ *
+ * @since 10.18.0
+ */
+
+@StatelessCheck
+public class MissingNullCaseInSwitchCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "missing.switch.nullcase";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.LITERAL_SWITCH};
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (!hasNullCaseLabel(ast) && (hasPatternCaseLabel(ast) || hasStringCaseLabel(ast))) {
+            log(ast, MSG_KEY);
+        }
+    }
+
+    /**
+     * Checks if the given switch AST node has a case null label.
+     *
+     * @param detailAST the AST node representing {@code LITERAL_SWITCH}
+     * @return true if the switch has {@code case null}, false otherwise
+     */
+    private static boolean hasNullCaseLabel(DetailAST detailAST) {
+        return TokenUtil.findFirstTokenByPredicate(detailAST, ast -> {
+            final DetailAST caseLabel = ast.findFirstToken(TokenTypes.LITERAL_CASE);
+            DetailAST expr = null;
+            if (caseLabel != null) {
+                expr = caseLabel.findFirstToken(TokenTypes.EXPR);
+            }
+            return expr != null && expr.findFirstToken(TokenTypes.LITERAL_NULL) != null;
+        }).isPresent();
```

Comment:
It is not clear to me why we need to lean so heavily on findFirstTokenByPredicate in this check. Can we just make some collection of  LITERAL_CASE DetailASTs, then simplify all of these predicates to only deal with one LITERAL_CASE at a time? Then in visitToken, we just do something like this:
```
        final boolean hasNullCase =  cases.stream()
          .filter(MissingNullCaseInSwitchCheck::hasLiteralNull)
          .isPresent();
```

Then, this method looks like:
```
return Optional.ofNullable(caseLabel.findFirsttoken(TokenTypes.EXPR)
    .map(expr -> expr.findFirstToken(TokenTypes.LITERAL_NULL)
    .isPresent();


```

Think about how we will extend this check in the future without refactoring. If we start with a collection of cases, and build check logic around this, I think it will be more maintainable going forward.

Reply:
This all all pseudocode obvs.

Reply:
> It is not clear to me why we need to lean so heavily on findFirstTokenByPredicate in this check.

no specific reason I just liked it and found it handy.

> Can we just make some collection of LITERAL_CASE DetailASTs, then simplify all of these predicates to only deal with one LITERAL_CASE at a time?

Good, but I think this will make the check Filestateful do we encourage this? I think somewhere else we stated that the check should always be stateless if possible

Reply:
>  It is not clear to me why we need to lean so heavily on findFirstTokenByPredicate in this check.

 No specific reason I just liked it and found it handy and also inspired by `MissingDefaultCheck`

> Can we just make some collection of LITERAL_CASE DetailASTs, then simplify all of these predicates to only deal with one LITERAL_CASE at a time?

Good, but I think this will make the check Filestateful do we encourage this? I think somewhere else we stated that the check should always be stateless if possible

Reply:
No, we don’t need to make this check stateful. Right now, we repeatedly iterate over all children of a given switch via findFirstTokenByPredicate. I am proposing we just make a collection instead.

Reply:
this `LITERAL_CASE` collection will be a class field, right? Isn't this a state

Reply:
>class field, right

Nope. Just make a List/Set/whatever makes sense in visitToken.

Reply:
See https://github.com/nrmancuso/checkstyle/blob/b8615479028fb40523376d6093db01df9db6876a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ConstructorsDeclarationGroupingCheck.java#L92 as an example.